vector-app/src/main/java/im/vector/app/core/di/SingletonModule.kt

```diff
@@ -168,6 +173,11 @@ import javax.inject.Singleton
         return Matrix(context, configuration)
     }
 
+    @Provides
```

Comment:
You should use `@Singleton ` otherwise listeners won't be kept.
Also use @Binds instead of @Provides

Reply:
Done in 6452b5c2b450da8de1f6ee4717b9940b49616b1e, hence I think it was fine since the listener was added and removed by the same entity (the Activity).
OK for you?